Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
-                } break;
-
-                // PIXMAN_r5g6b5
-                case _Color565: {
-                    unsigned short shortPixel = *reinterpret_cast<unsigned short*>(pixel);
-
-                    // Mask out the RGB portions
-                    outSwizzle->r = (BYTE)(shortPixel >> 11);
-                    outSwizzle->g = (BYTE)((shortPixel >> 5) & 0x3F);
-                    outSwizzle->b = (BYTE)(shortPixel & 0x1F);
-
-                    // Scale component to BYTE with LSB extension. (00011b << 3 becomes 11111b, 00010b becomes 10000b)
-                    outSwizzle->r = (outSwizzle->r << 3) | (((outSwizzle->r & 0x1) << 3) - (outSwizzle->r & 0x1));
-                    outSwizzle->g = (outSwizzle->g << 2) | (((outSwizzle->g & 0x1) << 2) - (outSwizzle->g & 0x1));
-                    outSwizzle->b = (outSwizzle->b << 3) | (((outSwizzle->b & 0x1) << 3) - (outSwizzle->b & 0x1));
-                } break;
-
-                // PIXMAN_a8r8g8b8
-                case _ColorARGB: {
-                    outSwizzle->r = pixelSwizzle->r;
-                    outSwizzle->g = pixelSwizzle->g;
-                    outSwizzle->b = pixelSwizzle->b;
-                    outSwizzle->a = pixelSwizzle->a;
-                } break;
-
-                // PIXMAN_x8b8g8r8 | PIXMAN_a8b8g8r8
-                case _ColorXBGR:
-                case _ColorABGR: {
-                    outSwizzle->r = pixelSwizzle->b;
-                    outSwizzle->g = pixelSwizzle->g;
-                    outSwizzle->b = pixelSwizzle->r;
-                    outSwizzle->a = pixelSwizzle->a;
-                } break;
-
-                // PIXMAN_b8g8r8x8
-                case _ColorBGRX: {
-                    outSwizzle->r = pixelSwizzle->b;
-                    outSwizzle->g = pixelSwizzle->g;
-                    outSwizzle->b = pixelSwizzle->r;
-                    outSwizzle->a = pixelSwizzle->a;
-                } break;
-
-                default:
-                    // Impossible state, we should have failed higher up.
-                    FAIL_FAST_HR(E_UNEXPECTED);
-                    break;
-            }
-            rowOut += xStrideOut;
-            rowStart += xStrideImg;
-        }
-        pImgData += yStrideImg;
-        png_write_row(png_ptr, pRow);
-    }
-    IwFree(pRow);
-    pImage->Backing()->ReleaseImageData();
-
-    png_write_end(png_ptr, NULL);
-    png_destroy_write_struct(&png_ptr, &info_ptr);
-
-    return ret;
+/**
+ @Status Stub
+ */
+CGImageRef CGImageCreateWithMask(CGImageRef image, CGImageRef mask) {
+  // Given how masks are applied during rendering via D2D, we will hold onto the
+  // mask then apply it at the appropriate time.
+  RETURN_NULL_IF(!image);
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
-@interface _UIImageWriterCallback : NSObject
-- (void)image:(UIImage*)image didFinishSavingWithError:(NSError*)err contextInfo:(void*)contextInfo;
-@end
-
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
```

Comment:
status will be updated in later iteration

---
In reply to: [84143387](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84143387) [](ancestors = 84143387)